include/mango_v3.hpp

```diff
@@ -197,6 +203,247 @@ struct EventQueue {
   AnyEvent items[EVENT_QUEUE_SIZE];
 };
 
+// todo: change to scoped enum class
+enum NodeType : uint32_t {
+  Uninitialized = 0,
+  InnerNode,
+  LeafNode,
+  FreeNode,
+  LastFreeNode
+};
+
+struct AnyNode {
+  NodeType tag;
+  uint8_t padding[BOOK_NODE_SIZE - 4];
+};
+
+struct InnerNode {
+  NodeType tag;
+  uint32_t prefixLen;
+  __uint128_t key;
+  uint32_t children[2];
+  uint8_t padding[BOOK_NODE_SIZE - 32];
+};
+
+struct LeafNode {
+  NodeType tag;
+  uint8_t ownerSlot;
+  uint8_t orderType;
+  uint8_t version;
+  uint8_t timeInForce;
+  __uint128_t key;
+  solana::PublicKey owner;
+  uint64_t quantity;
+  uint64_t clientOrderId;
+  uint64_t bestInitial;
+  uint64_t timestamp;
+};
+
+struct FreeNode {
+  NodeType tag;
+  uint32_t next;
+  uint8_t padding[BOOK_NODE_SIZE - 8];
+};
+
+struct Order {
+  Order(uint64_t price, uint64_t quantity) : price(price), quantity(quantity) {}
+  uint64_t price = 0;
+  uint64_t quantity = 0;
+};
+
+struct L1Orderbook {
+  uint64_t highestBid = 0;
+  uint64_t highestBidSize = 0;
+  uint64_t lowestAsk = 0;
+  uint64_t lowestAskSize = 0;
+  double midPoint = 0.0;
+  double spreadBps = 0.0;
+
+  bool valid() const {
+    return ((highestBid && lowestAsk) && (lowestAsk > highestBid)) ? true
+                                                                   : false;
+  }
+};
+
+class BookSide {
+ public:
+  BookSide(Side side) : side(side) {}
+
+  bool handleMsg(const nlohmann::json &msg) {
```

Comment:
this should really live in the websocket subscription code, rather than the account type. in fact the code should work the same for all types of account subscriptions.

Reply:
the first four lines to check json "result" or the whole of msg handling? 
first four lines could be moved one layer below to wss subscription code in case we know that each msg has the "result" object. 
the other parsing code must be handled by the account itself.